src/openvic-simulation/map/Map.cpp

```diff
@@ -529,21 +536,28 @@ bool Map::load_map_images(fs::path const& province_path, fs::path const& terrain
 					);
 				}
 			}
+
 			province_shape_image[idx].index = Province::NULL_INDEX;
 
-		set_terrain:
-			const TerrainTypeMapping::index_t terrain = terrain_data[idx];
-			TerrainTypeMapping const* mapping = terrain_type_manager.get_terrain_type_mapping_for(terrain);
-			if (mapping != nullptr) {
-				if (province_shape_image[idx].index != Province::NULL_INDEX) {
-					terrain_type_pixels_list[province_shape_image[idx].index - 1][&mapping->get_type()]++;
+			set_terrain:
+				const Province::index_t province_index = get_index_from_colour(province_colour);
+				const uint16_t array_index = province_index-1;
```

Comment:
You should check the province index isn't 0 before doing this, otherwise you'll wrap around and get a value of 65535. If the province index is 0, there's no centre to calculate so the code can just continue on to the terrain part below.

Reply:
I thought the code above checks for null index before entering set_terrain.

Reply:
I changed it to use idx as you suggested in another comment. So no more -1.

Reply:
If checks for null index before jumping ahead to set_terrain, but if no index can be found for the colour it continues on to set the index to NULL_INDEX just above set_terrain, and then continues on to the code under set_terrain.